src/javis_viewer.jl

```diff
@@ -0,0 +1,292 @@
+                    # Gets a Javis frame to display based on textbox entry
+                    frame_mat = transpose(get_javis_frame(video, action_list, frames))
+                    # Gets the correct Canvas context to draw on
+                    context = getgc(canvas)
+                    image(
+                        context,
+                        CairoImageSurface(frame_mat),
+                        0,
+                        0,
+                        frame_dims[1],
+                        frame_dims[2],
+                    )
+                end
+            end
+        elseif curr_frame < frames
+            if event.keyval == 65293
+                @guarded draw(canvas) do widget
+                    # Gets a Javis frame to display based on textbox entry
+                    frame_mat = transpose(get_javis_frame(video, action_list, 1))
+                    # Gets the correct Canvas context to draw on
+                    context = getgc(canvas)
+                    image(
+                        context,
+                        CairoImageSurface(frame_mat),
+                        0,
+                        0,
+                        frame_dims[1],
+                        frame_dims[2],
+                    )
+                end
+            end
+        end
+    end
+
+    # When the `forward` button is clicked, increment current frame number
+    signal_connect(forward, "clicked") do widget
+        curr_frame = parse(Int, get_gtk_property(tbox, :text, String))
+        if frames > curr_frame
+            push!(slide, curr_frame + 1)
+            @guarded draw(canvas) do widget
+                # Gets the next Javis frame based on textbox entry
+                frame_mat = transpose(get_javis_frame(video, action_list, curr_frame + 1))
+                # Gets the correct Canvas context to draw on
+                context = getgc(canvas)
+                image(
+                    context,
+                    CairoImageSurface(frame_mat),
+                    0,
+                    0,
+                    frame_dims[1],
+                    frame_dims[2],
+                )
+            end
+        else
+            push!(tbox, 1) # Sets the first frame shown to one
+            @guarded draw(canvas) do widget
+                # Gets the next Javis frame based on textbox entry
+                frame_mat = transpose(get_javis_frame(video, action_list, 1))
+                # Gets the correct Canvas context to draw on
+                context = getgc(canvas)
+                image(
+                    context,
+                    CairoImageSurface(frame_mat),
+                    0,
+                    0,
+                    frame_dims[1],
+                    frame_dims[2],
+                )
+            end
+        end
+    end
+
+    # When the `Right Arrow` key is pressed, increment current frame number
+    signal_connect(win, "key-press-event") do widget, event
```

Comment:
this is the same as clicking the `==>` button, right? Therefore make it a method :wink:

Reply:
It has turned into a method!

Reply:
not it hasn't. You still have twice the exact same code.

Reply:
Ah! I missed this. I have now properly modularized it